components/mediation-ui/mediators-ui/org.wso2.carbon.mediator.enrich.ui/src/main/java/org/wso2/carbon/mediator/enrich/ui/EnrichMediator.java

```diff
@@ -177,13 +178,23 @@ public OMElement serialize(OMElement parent) {
 
             if (sourceInlineXML != null && !sourceInlineXML.equals("")) {
                 OMElement inlineXMLElement;
+                JsonParser parser = new JsonParser();
+                // check for inline JSON values
                 try {
-                    inlineXMLElement = AXIOMUtil.stringToOM(sourceInlineXML);
-                    if (inlineXMLElement != null) {
-                        sourceElem.addChild(inlineXMLElement);
+                    parser.parse(sourceInlineXML);
+                    String formattedString = "<source xmlns=\"" + SynapseConstants.SYNAPSE_NAMESPACE +
+                            "\" type=\"inline\">" + sourceInlineXML + "</source>";
```

Comment:
AFAIK operator should precede the statement. Shall we fix?

Reply:
+1 fixed